pkg/examples/containerized-workload/main.go

```diff
@@ -0,0 +1,60 @@
+package main
```

Comment:
examples should not live in `pkg` in general. I'd suggest:

```
developing/sample_controller/main.go
```

Reply:
I would advocate for naming the top-level directory `examples/` if it is all the same to everyone. Another option would be to just point to Crossplane's usage of it as an example, then provide smaller pointers in documentation